romancal/resample/resample_utils.py

```diff
@@ -402,3 +419,29 @@ def resample_range(data_shape, bbox=None):
         ymax = min(data_shape[0] - 1, int(y2 + 0.5))
 
     return xmin, xmax, ymin, ymax
+
+
+def add_var_sky_array(input_models: ModelLibrary):
+    """
+    Add sky variance array to each model of a ModelLibrary.
+
+    Parameters
+    ----------
+    input_models : ModelLibrary
+        A library of models to which the sky variance array will be added.
+
+    Returns
+    -------
+    None
+    """
+    with input_models:
+        ref_img = input_models.borrow(index=0)
+        input_models.shelve(model=ref_img, index=0)
+        for i, img in enumerate(input_models):
+            if np.all(img.data != 0):
```

Comment:
Sorry, my fault.  This is good protection for the img.data = 0 case, but probably too good.

In an average image we should expect some pixels with flux == 0; say they're masked or something and are marked as DO_NOT_USE.  Then var_poisson will also be 0 or NaN or something.  We need to handle this case without giving up entirely.  Let's handle this for now via something like:
```
m = img.data != 0
img['var_sky'][m] = img.var_rnoise[m] + img.var_poisson[m] / img.data[m] * np.median(img.data)
img['var_sky'][~m] = img.var_rnoise[~m]
```
If we wanted to do better we should probably do only a median over the pixels that are "good", so something like
np.median(img.data[(img.dq & DO_NOT_USE) != 0]), for example.

Reply:
Done.